src/mongocxx/test/spec/unified_tests/operations.cpp

```diff
@@ -0,0 +1,1007 @@
+        auto& session = map.get_client_session(object);
+        session.commit_transaction();
+        return empty_doc;
+    }
+    if (name == "assertSessionTransactionState") {
+        auto session_name = op["arguments"]["session"].get_string().value.to_string();
+        auto& session = map.get_client_session(session_name);
+        return assert_session_transaction_state(session, op_view);
+    }
+    if (name == "dropCollection") {
+        auto coll_name = op["arguments"]["collection"].get_string().value.to_string();
+        auto& db = map.get_database(object);
+        db.collection(coll_name).drop();
+        return empty_doc;
+    }
+    if (name == "createCollection") {
+        auto coll_name = op["arguments"]["collection"].get_string().value.to_string();
+        auto& db = map.get_database(object);
+        if (op["arguments"]["session"]) {
+            auto session_name = op["arguments"]["session"].get_string().value.to_string();
+            auto& session = map.get_client_session(session_name);
+            db.create_collection(session, coll_name);
+        } else {
+            db.create_collection(coll_name);
+        }
+        return empty_doc;
+    }
+    if (name == "assertCollectionNotExists") {
+        REQUIRE_FALSE(collection_exists(op_view));
+        return empty_doc;
+    }
+    if (name == "assertCollectionExists") {
+        REQUIRE(collection_exists(op_view));
+        return empty_doc;
+    }
+    if (name == "createIndex") {
+        auto& coll = map.get_collection(object);
+        if (op["arguments"]["session"]) {
+            auto session_name = op["arguments"]["session"].get_string().value.to_string();
+            auto& session = map.get_client_session(session_name);
+            return create_index(coll, &session, op_view);
+        }
+        return create_index(coll, nullptr, op_view);
+    }
+    if (name == "assertIndexNotExists") {
+        REQUIRE_FALSE(index_exists(op_view));
+        return empty_doc;
+    }
+    if (name == "assertIndexExists") {
+        REQUIRE(index_exists(op_view));
+        return empty_doc;
+    }
+    if (name == "withTransaction") {
+        auto cb = with_transaction_cb{
+            array::value(op["arguments"]["callback"].get_array().value), map, apm};
+        auto& session = map.get_client_session(object);
+        return with_transaction(session, op_view, cb);
+    }
+    if (name == "delete") {
+        auto& bucket = map.get_bucket(object);
+        auto arguments = op["arguments"];
+        bucket.delete_file(arguments.get_document().view()["id"].get_value());
+        return empty_doc;
+    }
+    if (name == "download") {
+        auto& bucket = map.get_bucket(object);
+        return download(bucket, op_view);
+    }
+    if (name == "upload") {
+        return upload(map, op_view);
+    }
+    if (name == "deleteOne") {
+        auto& coll = map.get_collection(object);
+        return delete_one(coll, op_view);
+    }
+
```

Comment:
When we test against sharded clusters, we'll need to add `targetedFailpoint`. But I think it is ok to leave as future work, as long as we fail with the "unsupported operation" exception, which should be clear.